modules/phase_field/src/meshgenerators/SphereSurfaceMeshGenerator.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SphereSurfaceMeshGenerator.h"
+#include "CastUniquePointer.h"
+
+// libMesh includes
+#include "libmesh/face_tri3.h"
+#include "libmesh/mesh_modification.h"
+#include "libmesh/mesh_refinement.h"
+#include "libmesh/sphere.h"
+#include "libmesh/unstructured_mesh.h"
+#include "libmesh/replicated_mesh.h"
+
+registerMooseObject("PhaseFieldApp", SphereSurfaceMeshGenerator);
+
+template <>
+InputParameters
+validParams<SphereSurfaceMeshGenerator>()
+{
+  InputParameters params = validParams<MeshGenerator>();
+  params.addClassDescription(
+      "Generated sphere mesh - a two dimensional manifold embedded in three dimensional space");
+  params.addParam<Real>("radius", 1.0, "Sphere radius");
+  params.addParam<Point>("center", Point(0, 0, 0), "Center of the sphere");
+  params.addParam<unsigned int>(
+      "depth", 3, "Iteration steps in the triangle bisection construction");
+  return params;
+}
+
+SphereSurfaceMeshGenerator::SphereSurfaceMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _radius(getParam<Real>("radius")),
+    _center(getParam<Point>("center")),
+    _depth(getParam<unsigned int>("depth"))
+{
+}
+
+std::unique_ptr<MeshBase>
+SphereSurfaceMeshGenerator::generate()
+{
+  std::unique_ptr<ReplicatedMesh> mesh = libmesh_make_unique<ReplicatedMesh>(comm(), 2);
```

Comment:
This isn't the right line of code. This won't work with DistributedMesh runs or other `[Mesh]` block overrides. Do this instead:
https://github.com/idaholab/moose/blob/next/framework/src/meshgenerators/GeneratedMeshGenerator.C#L105-L106